packages/x-date-pickers/src/PickersLayout/usePickerLayout.tsx

```diff
@@ -117,10 +119,14 @@ const usePickerLayout = <TValue, TView extends DateOrTimeView>(
 
   // Tabs
 
-  const Tabs = components?.Tabs;
+  const Tabs = slots?.tabs ?? components?.Tabs;
   const tabs =
     view && Tabs ? (
-      <Tabs view={view} onViewChange={onViewChange} {...componentsProps?.tabs} />
+      <Tabs
+        view={view}
+        onViewChange={onViewChange}
+        {...(slotsProps?.tabs ?? componentsProps?.tabs)}
```

Comment:
Maybe going with the following pattern used in some other places would result in a cleaner result? 🤔 

```tsx
  ...
  components,
  componentsProps,
  slots,
  slotsProps: innerSlotsProps,
} = props;

const slotsProps = innerSlotsProps ?? componentsProps;
...
  externalSlotProps: slotsProps?.toolbar,
```